syncd/ComparisonLogic.cpp

```diff
@@ -2806,6 +2806,53 @@ void ComparisonLogic::cretePreMatchForLagMembers(
     }
 }
 
+void ComparisonLogic::cretePreMatchForAclEntries(
+        _In_ const AsicView& cur,
+        _Inout_ AsicView& tmp,
+        _Inout_ std::set<std::string>& processed)
+{
+    SWSS_LOG_ENTER();
+
+    // match acl entries which have the same priority attribute
+
+    auto cAclEntries = cur.getObjectsByObjectType(SAI_OBJECT_TYPE_ACL_ENTRY);
+    auto tAclEntries = tmp.getObjectsByObjectType(SAI_OBJECT_TYPE_ACL_ENTRY);
+
+    for (auto& cAclEntry: cAclEntries)
+    {
+        if (processed.find(cAclEntry->m_str_object_id) != processed.end())
+            continue;
+
+        for (auto& tAclEntry: tAclEntries)
+        {
+            if (processed.find(tAclEntry->m_str_object_id) != processed.end())
+                continue;
+
+            auto cPrio = cAclEntry->getSaiAttr(SAI_ACL_ENTRY_ATTR_PRIORITY);
+            auto tPrio = tAclEntry->getSaiAttr(SAI_ACL_ENTRY_ATTR_PRIORITY);
+
+            if (cPrio->getStrAttrValue() != tPrio->getStrAttrValue())
```

Comment:
Rookie question: are ACL priorities guaranteed to be always unique? If not, then we will likely run into same issue again.

Reply:
no they are not guaranteed in general, but sonic make sure that they are, since we want to control the right order of the acl, so having 2 acls with the same priority not guarantee the order of them